src/shared/elf-util.c

```diff
@@ -266,6 +266,42 @@ static int thread_callback(Dwfl_Thread *thread, void *userdata) {
         return DWARF_CB_OK;
 }
 
+static void report_module_metadata(StackContext *c, const char *name, JsonVariant *metadata) {
+        assert(c);
+        assert(name);
+        assert(metadata);
+
+        if (!c->f)
+                return;
+
+        const char
+                *build_id = json_variant_string(json_variant_by_key(metadata, "buildId")),
+                *type = json_variant_string(json_variant_by_key(metadata, "type")),
+                *package = json_variant_string(json_variant_by_key(metadata, "name")),
+                *version = json_variant_string(json_variant_by_key(metadata, "version")),
+                *arch = json_variant_string(json_variant_by_key(metadata, "architecture"));
+
+        fprintf(c->f, "Module %s", name);
+        if (package) {
+                /* The version is only meaningful with a package name. Skip the
+                 * version/arch fields if package is unknown. Architecture is
+                 * meaningful even without version, so print that. */
+                fprintf(c->f, " from %s %s", type ?: "package", package);
```

Comment:
would love some terminal color here, i.e. ansi_highligh() around the package name, as that is in 95% the most relevant bit of information here.

Reply:
This goes as text to the journal and must be plain text.